fe/fe-core/src/test/java/com/starrocks/lake/StarOSAgentTest.java

```diff
@@ -175,6 +175,36 @@ public void testAddAndRemoveWorker() throws Exception {
         Assert.assertEquals(-1, starosAgent.getWorkerIdByBackendId(5));
     }
 
+    @Test
+    public void testAddWillRemovePreviousWorker() throws Exception {
+        final String workerHost = "127.0.0.1:8090";
+        final long workerId1 = 10;
+        final long workerId2 = 11;
+        new Expectations() {
+            {
+                client.addWorker("1", workerHost);
+                minTimes = 0;
+                result = workerId1;
+            }
+        };
+
+        long backendId = 5;
+        Deencapsulation.setField(starosAgent, "serviceId", "1");
+        starosAgent.addWorker(backendId, workerHost);
+        Assert.assertEquals(workerId1, starosAgent.getWorkerIdByBackendId(backendId));
+
+        final String workerHost2 = "127.0.0.1:8091";
+        new Expectations() {
```

Comment:
add exepctation, expect following call will be made
```
client.removeWorker("1", workerId1)
```

Reply:
no need to do that, we can't trace the remove in test, i tested it manually